bears/general/KeywordBear.py

```diff
@@ -2,18 +2,67 @@
 
 from coalib.bearlib import deprecate_settings
 from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
 from coalib.results.Result import RESULT_SEVERITY, Result
 
+from bears.general.AnnotationBear import AnnotationBear
+from coalib.results.SourceRange import SourceRange
+
+
+def _get_comments(dependency_results):
+    if not dependency_results:
+        return {}
+
+    annotation_bear_results = dependency_results.get('AnnotationBear')
+    if (not annotation_bear_results or
+            not isinstance(annotation_bear_results, list)):
+        return {}
+
+    for result in annotation_bear_results:
+        yield from result.contents.get('comments', [])
```

Comment:
oh `_get_comments` is an iterator as you use `yield`. The above returned values have no effect, as these are not returned :)

So either return nothing above and keep `_get_comments` as a generator function (btw you return dicts above, I think you want sets, in this case `{}` --> `set()`), or change the last line to
```
return (result.contents.get('comments', []) for result in annotation_bear_results)
```